api/api_test.go

```diff
@@ -197,7 +205,9 @@ func TestServerReadOnlyRoutes(t *testing.T) {
 	if err != nil {
 		t.Fatal(err.Error())
 	}
-	s := New(node, cfg)
+	// TODO (b5) - hack until tests have better instance-generation primitives
+	inst := lib.NewInstanceFromConfigAndNode(cfg, node)
```

Comment:
Instead, once we switch to using the private `instance` struct for internal code, just change this to:

```
inst := &instance{config: cfg, node: node}
```

Reply:
because `api` imports `lib` this won't be possible because lib doesn't export `instance` or any of it's fields. I think in this case the right thing to do would be to use `lib.NewInstance`. `api` is a consumer of `lib`, it's `lib`'s job to make writing these tests straightforward.